### PR TITLE
SDK server not clearing lists on update

### PR DESCRIPTION
**Pull request by jlory**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

/kind bug

**What this PR does / Why we need it**:

Fixes the issue where items in list could not be removed.

**Which issue(s) this PR fixes**:

https://github.com/googleforgames/agones/issues/3593

**Special notes for your reviewer**:

Had to recreate the PR because of CLA issues.

### CODE REVIEW

File: pkg/sdkserver/sdkserver.go [hunk 1]
```diff
@@ -1230,8 +1230,8 @@ func (s *SDKServer) updateList(ctx context.Context) error {
 
 	// Cache a copy of the successfully updated gameserver
 	s.gsCopy = gs
-	// Clear the gsCounterUpdates
-	s.gsCounterUpdates = map[string]counterUpdateRequest{}
+	// Clear the gsListUpdates
+	s.gsListUpdates = map[string]listUpdateRequest{}
```
Comment (markmandel) [inline pkg/sdkserver/sdkserver.go:1234]: Do we need an update to a unit test to ensure this doesn't happen again?

Comment (jlory) [reply pkg/sdkserver/sdkserver.go:1234]: I'll have a look in the unit test file to see what's doable.

Comment (markmandel) [reply pkg/sdkserver/sdkserver.go:1234]: Appreciate you taking a look.

No rush, but just a heads up - we're cutting a release next Tuesday, so if you want this in the next release, we should get some unit tests in.

@igooch any advice on a good spot to write a unit test?

Comment (jlory) [reply pkg/sdkserver/sdkserver.go:1234]: I have something actually but it's at work, so tomorrow I should send an update to this MR, idea is:

- add a function that get the len() of the outgoing lists
- happy path = len() == 0
- network failure / issues updating = original len

Comment (igooch) [reply pkg/sdkserver/sdkserver.go:1234]: Looks good! The only thing I'd add is checking that the `gsListUpdates` has the update, i.e. is not zero after the update/add/remove call and before the get list call. Something like

```
if testCase.updated {
  updates := sc.gsListUpdatesLen()
  assert.Equal(t, 1, updates)
}
```

Comment (igooch) [reply pkg/sdkserver/sdkserver.go:1234]: ^Scratch that above suggestion, it would introduce a race condition. Code looks good as it stands.

**Review by igooch — APPROVED** [review]
LGTM

**Pull request merged**